ha/util/conf_store.py

```diff
@@ -110,3 +121,96 @@ def set_cluster_cardinality(self, index):
+            if cvg_count is None:
+                raise Exception(f"CVGs are not available for this node {node_id}")
+            cvg_list = [Conf.get(index, GconfKeys.CVG_NAME.value.format
+                                 (node_id=node_id, cvg_index=cvg_index)) for cvg_index in range(cvg_count)]
+            return cvg_list
+        except Exception as e:
+            Log.error(f"Unable to fetch CVG list from GConf. Error {e}")
+            raise Exception(f"Unable to fetch CVG list. Error {e}")
+
+    @staticmethod
+    def get_disk_list(index, node_id):
+        """
+        Return list of disks for any given node
+        Args:
+            index(str): index of conf file
+            node_id (str): node id
+        Returns:
+            list: list of disks id's
+
+        >>> ConftStoreSearch.get_disk_list("cortx", '21d6291109304485b3daff43a06cff77')
+        ['/dev/sdd', '/dev/sde', '/dev/sdc', '/dev/sdg', '/dev/sdh', '/dev/sdf']
+        """
+        disk_list = []
+        try:
+            cvg_count = Conf.get(index, GconfKeys.CVG_COUNT.value.format(node_id=node_id))
+            if cvg_count is None:
+                raise Exception(f"CVGs are not available for this node {node_id}")
+            for cvg_index in range(cvg_count):
+                num_of_data_disks = Conf.get(index, GconfKeys.DATA_COUNT.value.format(node_id=node_id, cvg_index=cvg_index))
+                if num_of_data_disks:
+                    for d_index in range(num_of_data_disks):
+                        disk_list.append(Conf.get(index, GconfKeys.DATA_DISK.value.
+                                                  format(node_id=node_id, cvg_index=cvg_index, d_index=d_index)))
+                num_of_metadata_disks = Conf.get(index, GconfKeys.METADATA_COUNT.value.format(node_id=node_id, cvg_index=cvg_index))
+                if num_of_metadata_disks:
+                    for m_index in range(num_of_metadata_disks):
+                        disk_list.append(Conf.get(index, GconfKeys.METADATA_DISK.value.
+                                                  format(node_id=node_id, cvg_index=cvg_index, m_index=m_index)))
+            return disk_list
+        except Exception as e:
+            Log.error(f"Unable to fetch Disk list from GConf. Error {e}")
+            raise Exception(f"Unable to fetch Disk list. Error {e}")
+
+    @staticmethod
+    def get_disk_list_for_cvg(index, node_id, cvg_id):
+        """
+        Return list of disks for any given node and CVG.
+        Args:
+            index(str): index of conf file
+            node_id (str): node id
+            cvg_id (str): cvg id
+        Returns:
+            list: list of disks id's
+
+        >>> ConftStoreSearch.get_disk_list_for_cvg("cortx", '21d6291109304485b3daff43a06cff77', 'cvg-01')
+        ['/dev/sdd', '/dev/sde', '/dev/sdc']
+        """
+        disk_list = []
+        try:
+            cvg_list = ConftStoreSearch.get_cvg_list(index, node_id)
+            if cvg_id in cvg_list:
```

Comment:
We can use the search APIs to get the full keys for the given CVG and values for the associated data/metadata disks right ?

Reply:
Discussed with akash. If we are to use search APIs, two iterations will have to be done. 
1. To get keys of all cvgs matching the cvg_id; across all nodes. ( there is no way to specify a particular node while searching for cvg_ids in search API)
2. To find out the correct key from 1 
3. To get disks associated with this CVG
so no benefit of using search APIs here.
hence closing the comment